runtime-transaction/src/extended_transaction.rs

```diff
@@ -0,0 +1,108 @@
+use {
+    solana_sdk::{
+        hash::Hash,
+        message::{v0::LoadedAddresses, SanitizedMessage},
+        signature::Signature,
+        transaction::{
+            Result, SanitizedTransaction, TransactionAccountLocks, VersionedTransaction,
+        },
+    },
+    std::time::Instant,
+};
+
+/// Sanitized transaction with optional start_time
+#[derive(Debug, Clone, Eq, PartialEq)]
+pub struct ExtendedSanitizedTransaction {
+    transaction: SanitizedTransaction,
+    start_time: Option<Instant>,
```

Comment:
I am not a huge fan of the extra memory per transaction since 99.9% of these will be None. It seems like we shouldn't be tacking this onto every transaction to me

Reply:
I think the extra memory usage is pretty limited. How much concurrent in memory transaction are we talking about?